pennylane/operation.py

```diff
@@ -1655,22 +1666,9 @@ def label(self, decimals=None, base_label=None, cache=None):
 
         return "@".join(ob.label(decimals=decimals) for ob in self.obs)
 
-    def queue(self, context=qml.QueuingContext, init=False):  # pylint: disable=arguments-differ
-        constituents = self.obs
-
-        if init:
-            constituents = self._args
-
-        for o in constituents:
-
-            if init:
-                if isinstance(o, Tensor):
-                    self.obs.extend(o.obs)
-                elif isinstance(o, Observable):
-                    self.obs.append(o)
-                else:
-                    raise ValueError("Can only perform tensor products between observables.")
+    def queue(self, context=qml.QueuingContext):  # pylint: disable=arguments-differ
```

Comment:
Why could we remove the `init` param and pull out the extend / append logic ?

Reply:
The extend/ append logic needs to be done *every* initialization, but queuing doesn't need to necessarily happen.  I think it was moved to the `queue` method so the code only need to perform one loop, but it breaks a clean separation of concerns between `__init__` and `queue`.